src/components/Layout/Header/ActionCenter/components/Notifications/SwapNotification.tsx

```diff
@@ -0,0 +1,116 @@
+import { CloseIcon } from '@chakra-ui/icons'
+import { Box, Button, Flex, HStack, Icon, Stack } from '@chakra-ui/react'
+import type { RenderProps } from '@chakra-ui/react/dist/types/toast/toast.types'
+import { useMemo } from 'react'
+import { TbArrowRight, TbCircleCheckFilled, TbCircleXFilled } from 'react-icons/tb'
+
+import { Amount } from '@/components/Amount/Amount'
+import { AssetIcon } from '@/components/AssetIcon'
+import { StreamIcon } from '@/components/Icons/Stream'
+import { SwapperIcon } from '@/components/MultiHopTrade/components/TradeInput/components/SwapperIcon/SwapperIcon'
+import { swapSlice } from '@/state/slices/swapSlice/swapSlice'
+import { useAppSelector } from '@/state/store'
+
+type SwapNotificationProps = {
+  handleClick: () => void
+} & RenderProps
+
+const toastHoverProps = {
+  transform: 'scale(1.02)',
+}
+
+export const SwapNotification = ({ handleClick, status, id, onClose }: SwapNotificationProps) => {
+  const swapsById = useAppSelector(swapSlice.selectors.selectSwapsById)
+
+  const swap = useMemo(() => {
+    if (!id) return undefined
+    return swapsById[id]
+  }, [id, swapsById])
+
+  const iconColor = status === 'success' ? 'green.500' : 'red.500'
+
+  if (!swap) return null
```

Comment:
e.g this and the above could be removed

Reply:
We can't do this, as the store doesn't know that it contains a swapId with this id, it can be undefined

But we could pass the entire swap to the render, the weird thing is that I'm not sure that it will be reactive, we have weird behaviors with `useColorModeValue` inside the render function of a toast, so I would probably just keep things as they are

this is pretty nitpick